pkg/autodiscovery/autoconfig.go

```diff
@@ -219,59 +233,15 @@ func (ac *AutoConfig) HasRunOnce() bool {
 }
 
 // GetAllConfigs queries all the providers and returns all the integration
-// configurations found, resolving the ones it can
+// configurations found.
 func (ac *AutoConfig) GetAllConfigs() []integration.Config {
-	return ac.getAllConfigs().schedule
-}
-
-// getAllConfigs queries all the providers and returns all the integration
-// configurations found, resolving the ones it can, and returns a configChanges to
-// schedule all of them.
-func (ac *AutoConfig) getAllConfigs() configChanges {
-	changes := configChanges{}
+	configs := []integration.Config{}
 
-	for _, pd := range ac.providers {
-		cfgs, err := pd.provider.Collect(context.TODO())
-		if err != nil {
-			log.Debugf("Unexpected error returned when collecting configurations from provider %v: %v", pd.provider, err)
-		}
-
-		if fileConfPd, ok := pd.provider.(*providers.FileConfigProvider); ok {
-			var goodConfs []integration.Config
-			for _, cfg := range cfgs {
-				// JMX checks can have 2 YAML files: one containing the metrics to collect, one containing the
-				// instance configuration
-				// If the file provider finds any of these metric YAMLs, we store them in a map for future access
-				if cfg.MetricConfig != nil {
-					// We don't want to save metric files, it's enough to store them in the map
-					ac.store.setJMXMetricsForConfigName(cfg.Name, cfg.MetricConfig)
-					continue
-				}
-
-				goodConfs = append(goodConfs, cfg)
-
-				// Clear any old errors if a valid config file is found
-				errorStats.removeConfigError(cfg.Name)
-			}
-
-			// Grab any errors that occurred when reading the YAML file
-			for name, e := range fileConfPd.Errors {
-				errorStats.setConfigError(name, e)
-			}
-
-			cfgs = goodConfs
-		}
-		// Store all raw configs in the provider
-		pd.overwriteConfigs(cfgs)
-
-		// resolve configs if needed
-		for _, config := range cfgs {
-			config.Provider = pd.provider.String()
-			changes.merge(ac.processNewConfig(config))
-		}
+	for _, pd := range ac.configPollers {
+		configs = append(configs, pd.getConfigs()...)
```

Comment:
Consider passing the slice to the method to avoid unnecessary copies: 
```
configs = pd.appendConfigs(configs)
```

Reply:
This code path is called from a single place (the agent's web UI, that I actually have never even seen 😅), and only once per interaction, so I don't think it's worth the change.